src/core/util/log.cc

```diff
@@ -131,6 +156,19 @@ void SetupLogging(const boost::program_options::variables_map& kovri_config)
   auto file_sink = boost::shared_ptr<text_file_sink>(
       std::make_unique<text_file_sink>(file_backend));
   // Set sink formatting
+  logging::formatter color_format = 
+      expr::stream << boost::phoenix::bind(&SetColor, logging::trivial::severity.or_none())
+                   << "["
+                   << expr::format_date_time(
+                          expr::attr<boost::posix_time::ptime>("TimeStamp"),
+                          "%Y.%m.%d %T.%f")
+                   << "]"
+                   << " [" << expr::attr<attrs::current_thread_id::value_type>(
+                                  "ThreadID")
+                   << "]"
+                   << " [" << logging::trivial::severity << "]"
+                   << "  " << expr::smessage
+                   << "\033[0m"; // Reset
   logging::formatter format =
```

Comment:
To eliminate the copypasta, one possible solution is to abstract the formatting process. Perhaps with a template function `template<typename Sink> SetFormatter(Sink& sink, bool is_file)` or local Boost.Variant object. Ideally, with either, the objective would be to test the sink object attributes (e.g., is it file or text sink?) internally so passing a bool would unnecessary.

There are probably more efficient solutions available though I don't know of any Boost.Log solutions off the top of my head (could be in the docs somewhere though).